src/Nethermind/Nethermind.Era1/EraModule.cs

```diff
@@ -31,15 +31,15 @@ protected override void Load(ContainerBuilder builder)
 
             // The admin export/import history method is here
             .RegisterSingletonJsonRpcModule<IEraAdminRpcModule, EraAdminRpcModule>()
-            ;
 
-        builder.RegisterBuildCallback((ctx) =>
-        {
-            IEraConfig eraConfig = ctx.Resolve<IEraConfig>();
-            if (string.IsNullOrWhiteSpace(eraConfig.NetworkName))
+            .AddDecorator<IEraConfig>((ctx, eraConfig) =>
```

Comment:
So that `ISpecProvider` is not always resolved. Taiko replace `ISpecProvider` with its own implementation.